### PR TITLE
docs: improvements to the code-sharing section

### PR DESCRIPTION
**Pull request by sebawita**
## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md#commit-messages.
- [ ] If there is an issue related with this PR, point it out here.

## What is the new state of the documentation article?
Added more content for the code-sharing docs

Fixes/Implements/Closes #[Issue Number].
n/a

### CODE REVIEW

File: docs/code-sharing/intro.md [hunk 1]
```diff
@@ -34,6 +34,175 @@ While, splitting the code for:
...
+* a **Basket Component**
+
+Like in this diagram:
+
+![basket-example](./img/basket.png?raw=true)
+
+The **BasketService** - should allow you to:
+
+* **loadItems** - get items placed in the basket
+* **updateItemQuantity** - allows to change the quantity of a given item in the basket
```
Comment (sis0k0) [inline docs/code-sharing/intro.md]: maybe remove `allows to` in order to follow the same format as the other items in the list.

File: docs/code-sharing/intro.md [hunk 2]
```diff
@@ -34,6 +34,175 @@ While, splitting the code for:
...
+* **remove** - remove a given item from the basket
+
+The **Basket Service** should be made of 100% shared code
```
Comment (sis0k0) [inline docs/code-sharing/intro.md]: shared -> shareable / platform-agnostic ?

Comment (sis0k0) [reply docs/code-sharing/intro.md]: Also add a dot at the end of the sentence.

File: docs/code-sharing/intro.md [hunk 3]
```diff
@@ -34,6 +34,175 @@ While, splitting the code for:
...
+}
+```
+
+**payment.service.ts**
+
+The **PaymentService** - should allow you to:
+
+* **processOneClickPayment** - process a payment using user's default card
+
+Again, the **PaymentService** should be made of 100% shared code
```
Comment (sis0k0) [inline docs/code-sharing/intro.md]: The same as the above comments.

File: docs/code-sharing/intro.md [hunk 4]
```diff
@@ -34,6 +34,175 @@ While, splitting the code for:
...
+
+@Injectable()
+export class PaymentService {
+  public processOneClickPayment() {
+    // Payment code
+  }
+}
+```
+
+Next, you would need to create a component, that would consume the **Basket** and the **Payment** services, and allow users to:
```
Comment (sis0k0) [inline docs/code-sharing/intro.md]: consume -> inject

File: docs/code-sharing/intro.md [hunk 5]
```diff
@@ -34,6 +34,175 @@ While, splitting the code for:
...
+
+The final bit is the implementation of the view templates.
+Both, templates should:
+
+* display a list of items - their name and quantity,
+* allow to increase or decrease the quantity of an item,
+* remove an item and
+* allow to process the payment
+
+In this case the UI for the web and mobile apps is going to be different, and as a result the template code **cannot be shared**. However, everything underneath that drives it's behaviour is still shared.
```
Comment (sis0k0) [inline docs/code-sharing/intro.md]: it's behaviour -> its behaviour

File: docs/code-sharing/intro.md [hunk 6]
```diff
@@ -34,6 +34,175 @@ While, splitting the code for:
...
+        <Button [text]="🗑" (tap)="removeItem(item)"></Button>
+      </GridLayout>
+    </ng-template>
+  </ListView>
+
+  <Button text="BUY NOW💰" (tap)="pay()"></Button>
+</StackLayout>
+```
+
+Even though, the contents of **basket.component.html** and **basket.component.tns.html** is not quite the same, you can see that in general they follow the same structure. Both use the **async pipe** to load the data from **items$** and both use the component methods (**increaseQuantity(item)**, **pay()**).
```
Comment (sis0k0) [inline docs/code-sharing/intro.md]: the contents ... is -> the contents ... are

**Review by sis0k0 — CHANGES_REQUESTED** [review]

**Review by sis0k0 — APPROVED** [review]

**Review by etabakov — APPROVED** [review]

**Pull request merged**